[Site Creation] Design Review

oguzkocer opened the pull request:
Fixes #8864 & #8883. @SylvesterWilmott and I have gone through a design review together on a Slack call and found several small issue to address. These issues have been documented in #8864 and addressed in this PR.  Here is a list of changes made in this PR:

* Adds consistent start & end margins to each site creation page using the `content_margin` which has different values depending on the device size. We were originally supposed to add a `maxWidth` but @SylvesterWilmott mentioned that he liked this approach by @theck13 in #8860 so we are going with that instead. It's also consistent with what the rest of the app does for tablet layouts.
* We decided to handle the missing `iconUrl` issue in the backend.
* Introduced styles for the common UI components and made sure they are consistent (I haven't done an exhaustive check, but I have extracted some bits and pieces as I found them)
* Fixed the widow word issues brought up by @theck13 [here](https://github.com/wordpress-mobile/WordPress-Android/pull/8863#pullrequestreview-187747414)
* We are now auto-focusing the first `EditText` in each page, but we are not opening the keyboard in the site info page and we are only opening it in domains page if the site doesn't have a title. The idea is to not block content when the user enters the screen. If the site has a title, we'll try to show domain suggestions, so showing the keyboard might potentially block content when the request is finished.
* Added a skeleton to the shimmer animation in the site preview page.
* Custom logic for showing the `Links are disabled` toast has been removed. It turns out this was working as intended before.
* When we receive the domain suggestions we'll show the results that matches the current query at the top of the list as a quality of life improvement.
* Updated ***Skip*** button margins as described in #8883.
* Verticals screen now has the same container as site info for the skip button
* Create site button in domains screen is now 36dp and it has a rounded corner similar to skip button in site info page
* Added shadow to the site url container in the site preview page

### Design Review

As mentioned, we have already gone through a design pass, but since we have a lot of fixes, I think it's a good idea to do it again. This will also serve as a documentation for all the designs that are implemented in this project.

First of here is me going through the site creation flow for different devices and orientations:

[Phone portrait](https://cloudup.com/c2PNlq3Fo_h)
[Phone landscape](https://cloudup.com/cCNmWcpnvdy)
[Tablet landscape](https://cloudup.com/comyoMDarNb)
[Tablet portrait](https://cloudup.com/cyXRkGTaf7E) _Sorry about the quality, the gif ended up being in wrong orientation and while fixing that I accidentally lowered the quality of it_

Here are the screenshots for successful flow:

[Phone portrait](https://cloudup.com/chBk0CXnhQr)
[Phone landscape](https://cloudup.com/cCHgnvKe6Yf)
[Tablet portrait](https://cloudup.com/cl8Uqwpk6pM)
[Tablet landscape](https://cloudup.com/cEe0PTDt1tI)

Here are screenshots for the errors:

[Phone portrait](https://cloudup.com/c-p_ED7dM1Y)
[Phone landscape](https://cloudup.com/cVD8t6aUK_d)
[Tablet portrait](https://cloudup.com/cDwne1EOBT8)
[Tablet landscape](https://cloudup.com/cTMo_b0VTVt)

### To test:

Going through the site creation flow following the recorded gifs for a phone and a tablet would probably be the best way to test this PR. We'll most likely have a separate PR to test the whole flow, so I think we can just focus on the changes of this PR which are many to list as testing instructions.

@theck13 I was wondering if you could check out the screenshots or the branch as well as another set of eyes? You don't have to review the code, but I'd feel more comfortable if you have gone through the flow at least once.

malinajirka commented:
Thanks @oguzkocer for all the changes:)

The code looks good. I've noticed few minor design issues.

1. IMHO the shadow of the skip button container is too small. It's barely visible and I think it's not obvious there is a content below it. (it looks +- ok on a computer screen, but at first I thought there isn't any shadow when I saw it on a phone).
https://cloudup.com/iOQlimGMDWd

2. The corner of the skip button container is rounded - it seems a bit weird to me. Wdyt?
https://cloudup.com/c5KU6UJ1ih6

3. The header subtitles still contain widow words 
 https://cloudup.com/icyDJPRlL30
https://cloudup.com/i9CaY3qGIEc

SideNote for @SylvesterWilmott - I've also noticed the horizontal margins on tablets are not consistent with refreshed stats https://cloudup.com/iG4cJrE08FE

SylvesterWilmott commented:
>IMHO the shadow of the skip button container is too small. It's barely visible and I think it's not obvious there is a content below it. (it looks +- ok on a computer screen, but at first I thought there isn't any shadow when I saw it on a phone).

The elevation applies a shadow that's offset in the Y axis so it appears below the item it's applied to. @oguzkocer can we affect the Y offset of the elevation or alternatively apply our own shadow?

>I've also noticed the horizontal margins on tablets are not consistent with refreshed stats

Nice spot @malinajirka . The margins are correct on this PR, @planarvoid pinging you so you're aware of this.

oguzkocer commented on WordPress/src/main/res/layout/new_site_creation_preview_skeleton_set.xml:
```diff
@@ -0,0 +1,33 @@
+        android:layout_marginEnd="@dimen/margin_extra_extra_large"/>
+
+    <View
+        android:layout_width="174dp"
```
The values were on purpose, but I forgot to create resources for them 🤦‍♂️ Fixed in e49ebf3d39e9e9d3e2731b7987eb7f1ad6d69258.

oguzkocer commented:
> > IMHO the shadow of the skip button container is too small. It's barely visible and I think it's not obvious there is a content below it. (it looks +- ok on a computer screen, but at first I thought there isn't any shadow when I saw it on a phone).

> The elevation applies a shadow that's offset in the Y axis so it appears below the item it's applied to. @oguzkocer can we affect the Y offset of the elevation or alternatively apply our own shadow?

@SylvesterWilmott @malinajirka I have increased the elevation from `2dp` to `6dp` in 6738cd50f4e5efe698c0c0022af9605d976da2b1. This is consistent with the elevation used for the `OK` button in the site preview page.

> The corner of the skip button container is rounded - it seems a bit weird to me. Wdyt?

I've removed the rounded corners in 6738cd50f4e5efe698c0c0022af9605d976da2b1.

> The header subtitles still contain widow words

In 06999ca194d9762fdef547a3023dd12d6f758cf1, I have refactored our views to use a single header layout and fixed the widow issues there. @malinajirka could you make sure to test the headers again since all of them are changed. Also, do you think we should use a more generic `tools:text` value or what we have in `new_site_creation_header_item.xml` is fine?

Here are a few updated screenshots: https://cloudup.com/crxq3qTG4C7. I don't want to go through every possibility again. Let me know if you need any other screenshots.

@SylvesterWilmott @malinajirka It should be ready for another look!

malinajirka commented:
Thanks @oguzkocer! I've tested it again on a phone and a tablet and it looks awesome!

>Also, do you think we should use a more generic tools:text value or what we have in new_site_creation_header_item.xml is fine?

I think it's fine as it is. 


I think it's ready for merge. Let me know when you review it @SylvesterWilmott ! Thanks

oguzkocer commented:
> Looking good, one minor thing, can we just make the next/skip button container 100% the screen width?

@SylvesterWilmott That might end up being hard to do. I notice this issue as well, but it's from the `content_margin` we set for the whole screen. If we do that, I think we might have to set the `content_margin` individually for every screen instead of as a whole. If that's the case, do you still want me to do it?

oguzkocer commented:
I have chatted with @SylvesterWilmott on Slack and it turns out I didn't fully understand what he meant. Initially I thought it was just the skip container, but he mentioned that vertical results should be full width as well. That means we didn't actually need to add individual margins, we just needed to change the `content_margin` value we were using just for phones and only on the portrait orientation. Long story short, the change ended up being a lot simpler: c198da1f1ded17dda339709ec445c4f092d78d2a

Here are some new screenshots just to showcase the margin difference (all other screens might have small fixes as well, but it'd take too long to take all of them again): https://cloudup.com/c_xrRZ4XyQ4 https://cloudup.com/c4ClETc_6ed

@SylvesterWilmott Are we all good now?

malinajirka commented:
LGTM, thanks!
